### PR TITLE
Add section on using the shell.

### PR DESCRIPTION
**Pull request by seanfisk**
**To review:**

1. Follow the instructions on [reviewing a pull request](http://seanfisk.com/mastering-eos/contributing/review-pull-request.html).
2. Follow through and test the entire section *Using the Command Line*.
3. *[Optional, but highly desired]* Follow through and test the entire section *User-level Software Installation*, which was modified as a result of the new shells section.

**Description:**

Completes much of issue #27:

- Available shells
  - Bash
  - sh
  - Zsh
  - fish
  - Others
  - Choosing a shell
  - Windows shells
    - Command Prompt
    - PowerShell
    - Cygwin
- Interactive shell use
  - Aliases
  - Built-ins
  - Quoting
  - Environment Variables
  - Pipelines
  - Redirection
  - Path Manipulation
  - Startup files
  - Utilities
  - Frameworks
- Shell scripting
  - Shebang
  - Boilerplate
  - When to shell script

The user-level software section and environment variables page were also modified to integrate with path manipulation and example dotfiles sections.

A script was added in order to test the example dotfiles.

### CODE REVIEW

**Comment by irawoodring** [comment]
- Would make it a bit clearer on the section on the environment, particularly the line
$ env | egrep '^GVSU'
that users will NOT see any output from this.

- Line
This example works only with Bash 4 (the default shell on EOS).
may be better as "Bash 4 and up"


-

File: manual/shell/index.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+========================
+ Using the Command Line
+========================
+
+*What is a shell?* A shell is an interface to a computer's operating system. You are probably already familiar with at least one shell: your machine's *graphical shell* , also know as a *graphical user interface* (GUI). This shell allows you to interact with windows, menus, buttons, and many other widgets using the keyboard, mouse, and other peripherals.
+
+Although there are shells of many types, the other most common type of shell is a text shell, also known as a *command-line interface* (CLI). Text shells are typically command interpreters: they wait for you to type commands, then execute them when you press :kbd:`Enter`. Interaction typically happens with the keyboard only. Unlike most graphical shells, text shells also excel at batch execution.
+
+Although text and graphical shells share some capabilities, each excels in unique ways. For many tasks, text shells represent a faster and more concise way to interact with the operating system. A successful computer scientist will learn to be comfortable using both graphical and text shells.
```
Comment (Liluye) [inline manual/shell/index.rst]: On first line "each excels". I think excels should be singular.

File: manual/shell/interactive.rst [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+After running the :cmd:`alias` command, :cmd:`hw` will now switch to that directory when typed.
+
+An alias can also be used to remember a lesser-used command. For example::
+
+   $ alias extract='tar -xvf'
+   $ extract bash-4.3.tar.gz
+   bash-4.3/
+   ...
+
+Notice that we can still pass arguments to the alias. In fact, it almost exactly like typing those characters at the command-line.
```
Comment (Liluye) [inline manual/shell/interactive.rst]: "In fact, it almost"  - it is almost

File: manual/shell/interactive.rst [hunk 3]
```diff
@@ -0,0 +1,692 @@
...
+   My current directory is: /home/smithj/awesome directory
+
+Double quotes are very frequently used, but it is easy to use them incorrectly. Know their behavior and test your commands with different values to make sure they are behaving correctly.
+
+.. _shell-env-vars:
+
+Environment Variables
+=====================
+
+:wikipedia:`A process' environment <Environment_variable>` is a mapping of key-value pairs possesed by every running process in the system. They are typically used to modify the behavior of programs. You are probably familiar with some common ones; for example, :envvar:`PATH`, :envvar:`EDITOR`, and :envvar:`PWD`. Environment variable names on Linux are case-sensitive and can contain most characters, although by convention they are usually named in all caps with words separated by underscores.
```
Comment (Liluye) [inline manual/shell/interactive.rst]: possessed misspelled.

File: manual/shell/interactive.rst [hunk 4]
```diff
@@ -0,0 +1,692 @@
...
+
+Utilities
+=========
+
+Various utilities can help streamline your use of the shell. Although they take some effort to install and use, the time saved by using them usually outweighs the setup cost.
+
+Directory Navigation
+--------------------
+
+One frequent task which can be expidited by utilities is changing directories. There are a few different tools which are popular for this task.
```
Comment (Liluye) [inline manual/shell/interactive.rst]: expedited misspelled.

**Comment by seanfisk** [comment]
@irawoodring 
> Would make it a bit clearer on the section on the environment, particularly the line `$ env | egrep '^GVSU'` that users will NOT see any output from this.

Fixed, thanks.

> This example works only with Bash 4 (the default shell on EOS).
> may be better as "Bash 4 and up"

I'm hesitant to do this because Bash 5 may change the syntax. I think that when Bash 5 comes out or the default shell changes, someone will have to manually update this.

File: manual/shell/index.rst (hunk 1 above)
Comment (seanfisk) [inline manual/shell/index.rst]: I used "excel" in the sentence before that too, so I just rephrased it :)

File: manual/shell/interactive.rst (hunk 2 above)
Comment (seanfisk) [inline manual/shell/interactive.rst]: Fixed.

File: manual/shell/interactive.rst (hunk 3 above)
Comment (seanfisk) [inline manual/shell/interactive.rst]: Fixed.

File: manual/shell/interactive.rst (hunk 4 above)
Comment (seanfisk) [inline manual/shell/interactive.rst]: Fixed.

**Comment by seanfisk** [comment]
All commented problems are fixed, except "Bash 4 and up" — see the comment.

File: manual/user-level-sw/user-hierarchies.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+   export INFOPATH=~/.local/share/info:$INFOPATH
+
+There is unfortunately some inconsistency with the location of installed man pages, which why we added both directories to the :envvar:`MANPATH`. :envvar:`INFOPATH` does not have these problems.
+
+Follow the directions for your session for :ref:`shell-startup-apply`. After this, the commands at the beginning of this section should bring up the correct documentation.
+
+Managing Paths
+==============
+
+Although the previous ``export`` commands work fine, there is a lot of repitition, especially if you would like to use :ref:`linuxbrew-section` as well. In this case, the following snippet from the :ref:`shell-example-files` is recommended as a replacement:
```
Comment (Liluye) [inline manual/user-level-sw/user-hierarchies.rst]: Typo, change "repitition" to repetition.

Comment (seanfisk) [inline manual/user-level-sw/user-hierarchies.rst]: Ugh, I need a decent spell-checker. Fixed :)

**Comment by Liluye** [comment]
Rechecked "Using the Command Line" and tested "User-level Software Installation". Marked one typo, other than that it looks good.

**Comment by seanfisk** [comment]
@Liluye Great, thanks! I'll wait for a :+1: from you and @irawoodring, but barring any other issues I think it's ready for merge!

**Comment by Liluye** [comment]
:+1:

**Comment by irawoodring** [comment]
:+1:  Nice job!

**Comment by seanfisk** [comment]
@irawoodring Thanks, much appreciated!

**Pull request merged**